angel_system/ptg_eval/common/discretize_data.py

```diff
@@ -3,21 +3,62 @@
 
 log = logging.getLogger("ptg_eval_common")
 
+
+def get_time_wind_range(start, end, dt, min_start_time, time_windows):
+        """
+        Return slice indices of time windows that reside completely in
+        start->end.
+
+        :param start: Timestamp for the start of the time window in seconds.
+        :param end: Timestamp for the end of the time window in seconds.
+        :param dt: Length of the time window in seconds.
+        :param min_start_time: The earliest timestamp between the
+            ground truth and detections.
+        :param time_windows: A list of all time windows in the format (start, end).
+
+        :return: Start and end indicies Tuple(ind1, ind2) where
+            time_windows[ind1:ind2] all live inside start->end.
+        """
+        # The start time of the ith window is min_start_time + dt*i.
+        ind1_ = (start - min_start_time)/dt
+        ind1 = int(np.ceil(ind1_))
+        if ind1_ - ind1 + 1 < 1e-15:
+            # We want to avoid the case where ind1_ is (j + eps) and it gets
+            # rounded up to j + 1.
+            ind1 -= 1
+
+        # The end time of the ith window is min_start_time + dt*(i + 1).
+        ind2_ = (end - min_start_time)/dt
+        ind2 = int(np.floor(ind2_))
+        if -ind2_ + ind2 + 1 < 1e-15:
+            # We want to avoid the case where ind1_ is (j - eps) and it gets
+            # rounded up to j - 1.
+            ind1 += 1
+
+        ind1 = max([ind1, 0])
+        ind2 = min([ind2, len(time_windows)])
+
+        return ind1, ind2
+
 def discretize_data_to_windows(labels, gt, detections, time_window, uncertainty_pad):
     """
-    :param labels: Array of class labels (str)
-    :param gt: Pandas dataframe of the ground truth
-    :param detections: Pandas dataframe of all detections per class
-    :param time_window: Time in seconds to discretize the whole time span by
+    Reformats the ground truth and detection data into time windows and 
+    removed any invalid time windows. 
+
+    :param labels: Array of class labels (str).
+    :param gt: Pandas dataframe of the ground truth.
+    :param detections: Pandas dataframe of all detections per class.
+    :param time_window: Time in seconds to discretize the whole time span by.
```

Comment:
The span of time covered by `detections` will be discretized in equally-long windows of this size (seconds).